hack/make-rules/verify.sh

```diff
@@ -74,6 +74,7 @@ QUICK_PATTERNS+=(
   "verify-gofmt.sh"
   "verify-imports.sh"
   "verify-non-mutating-validation.sh"
+  "verify-vanity-imports.sh"
```

Comment:
Important change: adds vanity imports to the `verify.sh`

Reply:
this is the main part I disagree with... I don't think vanity import comments are so important we should fail or block PRs on them

Reply:
I'm still -1 on making vanity imports a hard requirement that will fail pull requests. I don't think they are necessary or important enough to increase contributor friction over.

Reply:
I also think the autogeneration should be scoped to things under staging/src/k8s.io/...

Things under k8s.io/kubernetes/... outside of staging are not intended for use as a library.

Reply:
> If you check https://github.com/kubernetes/kubernetes/pull/120642/files#diff-e0ed342fde1ba5da0d2c7dcb506308e906b35c6b527c5150ec157da9b89b5e69R34-R35 we are just inspecting and showing a message, not making the PR fail. I can remove it too if you prefer so.

Ah, I missed that. Yeah, I don't think many people run this and inspect the results unless something in CI fails, let's drop this bit.